cobalt/browser/metrics/cobalt_enabled_state_provider_test.cc

```diff
@@ -0,0 +1,106 @@
+// Copyright 2025 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cobalt/browser/metrics/cobalt_enabled_state_provider.h"
+
+#include "base/command_line.h"
+#include "components/metrics/metrics_pref_names.h"
+#include "components/metrics/metrics_switches.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/testing_pref_service.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace cobalt {
+
+using metrics::prefs::kMetricsReportingEnabled;
+
+class CobaltEnabledStateProviderTest : public testing::Test {
+ protected:
+  CobaltEnabledStateProviderTest() {}
+
+  void SetUp() override {
+    pref_service_ = std::make_unique<TestingPrefServiceSimple>();
+    pref_service_->registry()->RegisterBooleanPref(kMetricsReportingEnabled,
+                                                   false);
+    state_provider_ =
+        std::make_unique<CobaltEnabledStateProvider>(pref_service_.get());
+    command_line_ = base::CommandLine::ForCurrentProcess();
+  }
+
+  TestingPrefServiceSimple* pref_service() { return pref_service_.get(); }
+  CobaltEnabledStateProvider* state_provider() { return state_provider_.get(); }
+  base::CommandLine* command_line() { return command_line_; }
```

Comment:
This is great; for future ref in GTests you can also make the members
protected and refer directly to them in the test cases (which are
subclasses).